chain/types/receipt.go

```diff
@@ -7,6 +7,7 @@ import (
 
 // MessageReceipt is the return value of message application.
 type MessageReceipt struct {
+	Msg         Message
```

Comment:
It looks like GasLImit and GasPrice are the only relevant properties, so you could reduce unnecessary coupling by only inlining those here. Would that allow revert of all the type changes above (I'm not sure if they were motivated by this).

Reply:
Unfortunately, I think we use the To, From, GasLimit, GasPrice, and Value.

I also don't think there's much wrong with embedding the Message into the MessageReceipt (there's good reason not to do so in the implementation itself, but I think it makes sense in a more limited sense such as chain-val).

Reply:
In my earlier comment I didn't realise that this was the MessageReceipt structure, intended to match the protocol spec structure exactly. I thought this was a higher level "evaluation result" structure, specific to chain validation.

We might need to introduce such a structure to solve the problem here.

Edit: we should not include the Value here.  That makes the tests way too magic and hard to follow. So that one should be excluded anyway on other grounds.

`From` and `To` are only used for some obscure check in the gas calculator that I don't understand. If I understood what it was attempting, we could probably find a better way to reach the same outcome.